oslo_vmware/vim_util.py

```diff
@@ -679,24 +679,118 @@ def is_vim_instance(o, vim_type_name):
                                                      sudsobject.Object))
 
 
-def serialize_object(obj):
-    """Convert Suds object into a dict."""
+def _get_child_type(sxtype, name):
+    if not sxtype:
+        return None
+    child, _ = sxtype.get_child(name)
+    return child.type[0]
+
+
+def serialize_object(obj, typed=False, expected_type=None):
+    """Serialize an SOAP object to a dict
+
+    For the function to serializes it in a format that the corresponding
+    deserialize_object can restore the original object, it needs to get
+    get passed typed=True, and the expected_type as string for polymorphism.
```

Comment:
A few typos here, too. `to serialize`, `get get`.

Reply:
Fixed